ApplicationStartup.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace poc_windowsservice_preflightcheck
+{
+    public static class ApplicationStartup
+    {
+        public static StartupMode GetStartupMode()
+        {
+            try
+            {
+                var options = new CommandLineOptions();
+                var isValid = CommandLine.Parser.Default.ParseArguments(Environment.GetCommandLineArgs(), options);
+
+                if (isValid && options.Preflight)
+                    return StartupMode.RunPreflightCheck;
+            }
+            catch(Exception e)
+            {
+                Console.WriteLine("Encountered an exception trying to determine StartupMode [RunNormally, RunPreflightCheck].");
+                Console.WriteLine(e.ToString());
```

Comment:
Maybe also add `Console.Error.WriteLine(options.GetUsage());` to show the options, that way you don't have to specify that yourself (it gets them from the Options class.)